rocAL/rocAL/source/meta_data/meta_node_resize_mirror_normalize.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright (c) 2019 - 2023 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "meta_node_resize_mirror_normalize.h"
+
+void ResizeMirrorNormalizeMetaNode::initialize()
+{
+    _src_height_val.resize(_batch_size);
+    _src_width_val.resize(_batch_size);
+    _dst_width_val.resize(_batch_size);
+    _dst_height_val.resize(_batch_size);
+    _mirror_val.resize(_batch_size);
+}
+void ResizeMirrorNormalizeMetaNode::update_parameters(pMetaDataBatch input_meta_data, pMetaDataBatch output_meta_data)
+{
+    initialize();
+    if(_batch_size != input_meta_data->size())
+    {
+        _batch_size = input_meta_data->size();
+    }
+    _mirror = _node->return_mirror();
+    auto input_roi = _node->get_src_roi();
+    auto output_roi = _node->get_dst_roi();
+    vxCopyArrayRange((vx_array)_mirror, 0, _batch_size, sizeof(uint), _mirror_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+
+    for (int i = 0; i < _batch_size; i++)
+    {
+        _dst_to_src_width_ratio = float(output_roi[i].x2) / float(input_roi[i].x2);
+        _dst_to_src_height_ratio = float(output_roi[i].y2) / float(input_roi[i].y2);
+
+        auto bb_count = input_meta_data->get_labels_batch()[i].size();
+        BoundingBoxCords coords_buf = input_meta_data->get_bb_cords_batch()[i];
+        Labels labels_buf = input_meta_data->get_labels_batch()[i];        
+        BoundingBoxCords bb_coords;
+        Labels bb_labels;
+        if (input_meta_data->get_metadata_type() == MetaDataType::PolygonMask)
+        {
```

Comment:
Only RMN has mask metadata modifications?

Reply:
Yes. Currently only added the mask updation in RMN node alone.